File: src/main/java/fm/comunas/ddd/aggregate/model/AggregateMeta.java

package fm.comunas.ddd.aggregate.model;

import fm.comunas.ddd.aggregate.model.enums.CodeAggregateType;
import fm.comunas.ddd.entity.model.EntityMeta;
import fm.comunas.ddd.validation.model.AggregatePartValidation;

import java.util.List;

/**
 * A DDD Aggregate Root Meta Information.
 */
public interface AggregateMeta extends EntityMeta {

	CodeAggregateType getAggregateType();

	List<AggregatePartValidation> getValidationList();

}

File: fm-dddrive/src/main/java/dddrive/ddd/core/model/AggregateMeta.kt
package dddrive.ddd.core.model

import dddrive.ddd.validation.model.AggregatePartValidation
import io.dddrive.oe.model.ObjUser
import java.time.OffsetDateTime

/**
 * A DDD Aggregate Root Meta Information.
 */
interface AggregateMeta : EntityMeta {

	val repository: AggregateRepository<*>

	val version: Int

	val validations: List<AggregatePartValidation>

	val createdAt: OffsetDateTime

	val createdByUser: ObjUser

	val modifiedAt: OffsetDateTime?

	val modifiedByUser: ObjUser?

}
